internal/eventtracker/eventtracker.go

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventtracker
+
+import (
+	"context"
+	"os"
+	"runtime"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/internal/attestation/crafter"
+	"github.com/denisbrodbeck/machineid"
+	"github.com/posthog/posthog-go"
+	"github.com/rs/zerolog"
+)
+
+// Tags represents a collection of event tags.
+type Tags map[string]string
+
+// TrackerClient defines the interface for tracking events.
+type TrackerClient interface {
+	TrackEvent(ctx context.Context, eventName string, tags Tags) error
+}
+
+// CommandTracker is a basic implementation of the Tracker interface.
+type CommandTracker struct {
+	client TrackerClient
+}
+
+// NewCommandTracker creates a new CommandTracker instance.
+func NewCommandTracker(client TrackerClient) *CommandTracker {
```

Comment:
commandTracker is going to need to implement a Track or Emit method no?

Reply:
That's what we had before an interface with a `Emit` method, on this way it would be something like, exposing the client:
```golang
tr := NewCommandTracker(posthogClient)
tr.Client.TrackEvent(...)
```

Reply:
Having a `TrackEvent` interface for the `Tracker`, can result on the following implementation on the CLI:


```golang
PersistentPostRun: func(cmd *cobra.Command, args []string) {
	var clientTracker eventtracker.Client
	if os.Getenv(doNotTrackEnv) == "true" || os.Getenv(doNotTrackEnv) == "1" {
		clientTracker = eventtracker.NewNoopClient()
	} else {
		clientTracker = eventtracker.NewPosthogClient(posthogAPIToken, posthogServerURL)
	}

	var cmdHierarchy []string
	evalCmd := cmd
	for evalCmd.Use != "chainloop" {
		cmdHierarchy = append(cmdHierarchy, evalCmd.Use)
		evalCmd = evalCmd.Parent()
	}

	slices.Reverse(cmdHierarchy)

	err := clientTracker.TrackEvent(cmd.Context(), "command_executed", eventtracker.Tags{
		"command": strings.Join(cmdHierarchy, " "),
	})
	if err != nil {
		logger.Debug().Err(err).Msg("failed to track event")
	}
},
```

Which I believe it's what we want.